lib/engine/train.rb

```diff
@@ -10,7 +10,7 @@ class Train
     attr_accessor :obsolete, :operated, :events
     attr_reader :available_on, :name, :distance, :discount, :obsolete_on,
                 :rusts_on, :rusted, :sym, :variant, :variants
-    attr_writer :buyable
+    attr_writer :buyable, :index
```

Comment:
yea i think you should get rid of this and just create the train in buy_company

Reply:
ok